File: app/src/main/java/maderski/bluetoothautoplaymusic/Utils/ReceiverUtils.java
package maderski.bluetoothautoplaymusic.Utils;

import android.content.ComponentName;
import android.content.Context;
import android.content.pm.PackageManager;
import android.util.Log;

/**
 * Created by Jason on 3/10/17.
 */

public class ReceiverUtils {
    private static final String TAG = "ReceiverHelper";

    // Start a receiver
    public static void startReceiver(Context context, Class<?> receiverClass) {
        Log.d(TAG, receiverClass.getSimpleName() + " STARTED!");
        try {
            ComponentName receiver = new ComponentName(context, receiverClass);
            PackageManager packageManager = context.getPackageManager();
            packageManager.setComponentEnabledSetting(receiver, PackageManager.COMPONENT_ENABLED_STATE_ENABLED, PackageManager.DONT_KILL_APP);
        }catch(Exception e){
            Log.e(TAG, e.getMessage());
        }
    }

    // Stop a receiver
    public static void stopReceiver(Context context, Class<?> receiverClass){
        Log.d(TAG, receiverClass.getSimpleName() + " STOPPED!");
        try {
            ComponentName btStateReceiver = new ComponentName(context, receiverClass);
            PackageManager packageManager = context.getPackageManager();
            packageManager.setComponentEnabledSetting(btStateReceiver, PackageManager.COMPONENT_ENABLED_STATE_DISABLED, PackageManager.DONT_KILL_APP);
        }catch (Exception e){
            Log.e(TAG, e.getMessage());
        }
    }
}

File: app/src/main/java/maderski/bluetoothautoplaymusic/utils/ReceiverUtils.kt
package maderski.bluetoothautoplaymusic.utils

import android.content.ComponentName
import android.content.Context
import android.content.pm.PackageManager
import android.util.Log

/**
 * Created by Jason on 3/10/17.
 */

object ReceiverUtils {
    private val TAG = "ReceiverHelper"

    // Start a receiver
    fun startReceiver(context: Context, receiverClass: Class<*>) {
        Log.d(TAG, receiverClass.simpleName + " STARTED!")
        try {
            val receiver = ComponentName(context, receiverClass)
            val packageManager = context.packageManager
            packageManager.setComponentEnabledSetting(receiver, PackageManager.COMPONENT_ENABLED_STATE_ENABLED, PackageManager.DONT_KILL_APP)
        } catch (e: Exception) {
            Log.e(TAG, e.message)
        }

    }

    // Stop a receiver
    fun stopReceiver(context: Context, receiverClass: Class<*>) {
        Log.d(TAG, receiverClass.simpleName + " STOPPED!")
        try {
            val btStateReceiver = ComponentName(context, receiverClass)
            val packageManager = context.packageManager
            packageManager.setComponentEnabledSetting(btStateReceiver, PackageManager.COMPONENT_ENABLED_STATE_DISABLED, PackageManager.DONT_KILL_APP)
        } catch (e: Exception) {
            Log.e(TAG, e.message)
        }

    }
}
